I just clicked the button, but it's still picking up: Successfully set up CPython (3.14.3)

Under

https://github.com/pybind/pybind11/actions/runs/24187486767/job/70595813802?pr=6032

I see:

```
  =================================== FAILURES ===================================
  ______________________________ test_get_referrers ______________________________
   
      def test_get_referrers():
          instance = m.DynamicAttr()
          instance.a = "test"
  >       assert instance in gc.get_referrers(instance.__dict__)
  E       AssertionError: assert <pybind11_tests.class_.DynamicAttr object at 0x7195a1b995e0> in []
  E        +  where [] = <built-in function get_referrers>({'a': 'test'})
  E        +    where <built-in function get_referrers> = gc.get_referrers
  E        +    and   {'a': 'test'} = <pybind11_tests.class_.DynamicAttr object at 0x7195a1b995e0>.__dict__
   
  instance   = <pybind11_tests.class_.DynamicAttr object at 0x7195a1b995e0>
   
  tests/test_class.py:52: AssertionError
```

I think it'll be best if you skip that test for Python versions between 3.14.0 and 3.14.3 (or similar; I'm not sure).

--- a/include/pybind11/detail/class.h
+++ b/include/pybind11/detail/class.h
@@ -578,9 +578,9 @@ inline PyObject *make_object_base_type(PyTypeObject *metaclass) {
 /// dynamic_attr: Allow the garbage collector to traverse the internal instance `__dict__`.
 extern "C" inline int pybind11_traverse(PyObject *self, visitproc visit, void *arg) {
 #if PY_VERSION_HEX >= 0x030D0000
-    int vret = PyObject_VisitManagedDict(self, visit, arg);
-    if (vret) {
-        return vret;
+    int ret = PyObject_VisitManagedDict(self, visit, arg);
+    if (ret) {
+        return ret;
     }
 #else
     PyObject *&dict = *_PyObject_GetDictPtr(self);

--- a/tests/test_class.py
+++ b/tests/test_class.py
@@ -19,6 +19,13 @@ def refcount_immortal(ob: object) -> int:
     return sys.getrefcount(ob)
 
 
+MANAGED_DICT_GET_REFERRERS_SUPPORTED = (
+    env.CPYTHON
+    and sys.version_info >= (3, 13, 13)
+    and (sys.version_info < (3, 14) or sys.version_info >= (3, 14, 4))
+)
+
+
 def test_obj_class_name():
     expected_name = "UserType" if env.PYPY else "pybind11_tests.UserType"
     assert m.obj_class_name(UserType(1)) == expected_name
@@ -46,6 +53,10 @@ def test_instance(msg):
     assert cstats.alive() == 0
 
 
+@pytest.mark.skipif(
+    not MANAGED_DICT_GET_REFERRERS_SUPPORTED,
+    reason="Requires CPython 3.13.13+ or 3.14.4+ managed dict traversal support",
+)
 def test_get_referrers():
     instance = m.DynamicAttr()
     instance.a = "test"

